numpy/ma/core.py

```diff
@@ -1224,6 +1227,177 @@ def __str__(self):
 fmod = _DomainedBinaryOperation(umath.fmod, _DomainSafeDivide(), 0, 1)
 mod = _DomainedBinaryOperation(umath.mod, _DomainSafeDivide(), 0, 1)
 
+# Include masked dot here as it is part of the __numpy_ufunc__ system. This
```

Comment:
I don't think `dot` is part of the ufunc system *yet*, though there are hopes for it (see #4164). 

I'd be slightly inclined to move the definition of `dot` to be just after the definitions of `inner` and `outer` near the bottom of the file.